### ISSUE TITLE
scope: copy paste

### ISSUE THREAD

**Issue by ssj71** [puglfork/pugl#4]
similar to #2 there is a PR for copy pase which I believe is fully cross-platform. http://dev.drobilla.net/ticket/1166

This relates to #3 but I have several use cases of plugins with text entry forms. Especially I could see a file selection benefit from copy paste. So I'd like to see this feature merged.

As always, please discuss.

**Comment by ventosus** [comment]
I definitely need copy/paste support for text fields in e.g. moony [1] (that's why I've written the above code).

btw, my pugl changes are here [2] (cannot push my 'omk' branch to this repo).

[1] https://open-music-kontrollers.ch/lv2/moony/
[2] https://github.com/ventosus/pugl/tree/omk

**Comment by ventosus** [comment]
Thinking about it, may be clipboard support can also be implemented separately for those that need it, instead of embedding it into pugl itself.

To be able to do so, one would need to be able to have access to

* a HWND on windows
* a Display on X11
* a NSwhatever on MacOS

If pugl's API allows to get access to the needed platform specifics, an external implementation should be doable. Maybe it is already for windows/x11, not so sure for MacOS, though

**Comment by ssj71** [comment]
how about https://github.com/puglfork/pugl/blob/master/pugl/pugl.h#L476 ?

I can definitely get behind the idea of keeping things as minimal as possible but providing easy reciepies for these sort of things. If its not an easy recipie, then there's a lot stronger argument for baking it into pugl.

**Comment by ventosus** [comment]
Yes, that should work for windows/x11, at least. Will try to decouple and report back here...

**Comment by ventosus** [comment]
copy-paste has landed in upstream [1]. I'll switch to use the latter, which makes my fork superfluous.

https://gitlab.com/lv2/pugl/commit/4a8cb07ced7c0d4268fa76e63f58496356ec339c
